panics.go

```diff
@@ -20,33 +20,28 @@ import (
 //
 // 	* Add an internal Error logger to report write errors, timeouts, or
 // 	  any other error that we can't directly surface to the user
-// 	* Make WriteTimeout configurable
 
 // Explain the reasoning behind NotifyContext, which provides a context
 // that is cancelled on any panic. If propagated then this will cancel
 // any inflight requests/work helping to prepare for exit.
 
-// WriteTimeout is the maximum amount of time to wait for a panic to be written
-// to output before handling the panic and signalling any registered channels
-// or canceling any registered contexts.
-//
-// This prevents a slow or blocked writer from blocking the handling of panics.
-const WriteTimeout = 100 * time.Millisecond
-
 var (
 	noPrintTrace   atomic.Bool
 	allStackTraces atomic.Bool
 	panicCount     atomic.Uint64 // number of captured panics
 	delivering     atomic.Int32  // panics being actively handled/delivered
 	firstPanic     atomic.Pointer[Error]
 	output         atomic.Pointer[writer]
+	writeTimeout   atomic.Int64
 )
 
 var stderrWriter = writer{w: os.Stderr}
 
 // immediately initialize output before any init() functions run
 var _ = func() struct{} {
 	output.Store(&stderrWriter)
+	// Initialize writeTimeout default to 100ms
+	writeTimeout.Store(int64(100 * time.Millisecond))
```

Comment:
Two things:

1. Let's declare a constant `DefaultWriteTimeout` for the sake of documentation.
2. After some consideration, I think we should increase the default write timeout to 500ms. When [`GOTRACEBACK`](https://pkg.go.dev/runtime#hdr-Environment_Variables) or [`debug.SetTraceback`](https://pkg.go.dev/runtime/debug#SetTraceback) is set to "all" (or greater) I've seen services with a large number of goroutines consistently produce stack traces that +20Mb in size. If a log ingestor is configured (say [fluent-bit](https://github.com/fluent/fluent-bit)) processing a 20Mb stack trace with a 100ms timeout would require a throughput of 200Mb/s whereas a 500ms timeout would only require 40Mb/s.

**Side note:** The more I think about it, the less convinced I am that this should be the default behavior and that it should really only serve as an optional backup and that it should maybe only print the first panic.

Reply:
Thanks for making the change and let's leave this comment unresolved for future reference (i.e. someone looking up why this value is what it is).